enforcer/src/daemon/engine.c

```diff
@@ -494,13 +502,28 @@ engine_setup(void)
                         if (fd > 2) (void)close(fd);
                     }
                     engine->daemonize = 0; /* don't fork again on reload */
+                    close(pipefd[0]);
                     break;
                 default: /* parent */
-                    exit(0);
+                    close(pipefd[1]);
+                    read(pipefd[0], &buff, 1);
```

Comment:
If this read fails buff will be 0, success. But I think it is better to assume failure.